js/src/common/utils/Drawer.js

```diff
@@ -31,7 +31,18 @@ export default class Drawer {
    * @public
    */
   hide() {
-    $('#app').removeClass('drawerOpen');
+    const $app = $('#app');
+
+    // If the drawer isn't open, stop execution
+    if (!$app.hasClass('drawerOpen')) return;
+
+    const $drawer = $('#drawer');
+
+    // Used to prevent `visibility: hidden` from breaking the exit animation
+    $drawer.one('transitionend', () => $drawer.css('visibility', ''));
+    $drawer.css('visibility', 'visible');
```

Comment:
I see that this works, but why are we setting the visibility to visible?  If that's what the above comment refers to, maybe likes 42 and 43 should be swapped?

Reply:
As soon as the drawer starts leaving the screen, the `drawerOpen` class is removed, which causes the `visibility: hidden` to get applied from here:

https://github.com/flarum/core/pull/2666/files#diff-cb687979202880be4ab817b086176117563c352bf18c053b6c9e6370ede80a0eR126-R128

`visibility: hidden` means that the close animation will stop as it hides the element. So, before we remove the class, we set `visibility: visible` to override that style, then we use `ontransitionend` to remove this when the drawer is hidden.

Reply:
I understand what you mean here, but I think it would be more readable if it was set to visible first, and then the animation one callback applied. Or even better, if we chained them (which I believe is possible)